src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
here you have sevral issues:
1. braces on one argument of arrow anonymous function (nees to be without braces)
2. you need to put yur code of arrow function iside curly braces (like in post link, that I added above)
3. method `find` returns `something that accepts your condition` or `undefined` and your logic will fall in case of `undefined.born` you need to check this issue. As a decision you can write `<your find function>?.born` - in this case `?` will check all undefined issues and your code won't fall.